tomatic/minizinc.py

```diff
@@ -0,0 +1,182 @@
+            maxTorns=self.maxTorns,
+            nTorns=self.nTorns,
+            indisponibilitats=self.indisponibilitats,
+            preferencies=self.preferencies
+        )
+
+    def translate(self, solution, config):
+        days = list(self.NORMAL_WEEKDAY.keys())
+        timetable = {
+            day: [
+                [
+                    self.NINGU for _ in range(self.nLinies)
+                ] for _ in range(self.nSlots)
+            ] for day in days
+        }
+
+        for day, turns in zip(self.laborable_days, solution.solution.ocupacioSlot):
+            for turn_i, turn in enumerate(turns):
+                for slot_i, person in enumerate(turn):
+                    timetable[day][turn_i][slot_i] = self.names[person-1]
+
+        result = ns(
+            week=f'{config.monday}',
+            days=days,
+            hours=config.hours,
+            turns=[ f"T{torn+1}" for torn in range(config.nTelefons) ],
+            timetable=timetable,
+            colors=config.colors,
+            extensions=config.extensions,
+            names=config.names,
+            overload={},  # empty here
+            penalties=[],  # empty here
+            cost=solution.solution.totalTorns,
+            log=[]  # TODO: empty?
+        )
+        return result
+
+
+def make_html_file(solution, filename):
+    html_gen = HtmlGen(solution)
+    with open(filename, 'w') as file:
+        file.write(
+            html_gen.htmlHeader() +
+            html_gen.htmlColors() +
+            html_gen.htmlSubHeader() +
+            html_gen.htmlSetmana() +
+            html_gen.htmlTable()+
+            html_gen.htmlPenalties()+
+            html_gen.htmlExtensions()+
+            html_gen.htmlFooter()
+        )
+
+
+def solve_problem(config, solvers):
+    menu = Menu(config)
+    # define a problem
+    tomatic_problem_params = menu.ingredients()
+    tomatic_problem = TomaticProblem(**tomatic_problem_params)
+    # create an instance of the cooker
+    tomato_cooker = GrillTomatoCooker(tomatic.MODEL_DEFINITION_PATH, solvers)
+    # Now, we can solve the problem
+    solution = asyncio.run(tomato_cooker.cook(tomatic_problem))
+    return menu.translate(solution, config) if solution else False
+
+
+def main():
+    global args
+    args = parseArgs()
+    config = Config(
+        args.config_file,
+        args.date,
+        args.keep,
+        args.certificate,
+        args.holidays
+    )
+    # TODO: check where to save this
+    target_date = args.date or config.data.monday
+    output_yaml = "graella-telefons-{}.yaml".format(target_date)
+    output_html = "graella-telefons-{}.html".format(target_date)
+    status_file = "status.yaml"
+    # Fist try to get a solution with optional absences
+    step('Provant amb les indisponibilitats opcionals...')
+    # choose a list of minizinc solvers to user
+    solvers = ["chuffed", "coin-bc"]
+    solution = solve_problem(config.data, solvers)
+    if not solution:
```

Comment:
May be this "if" can be inside `solve_problem` function or a `@retry` decorator

Reply:
About this, as we need to change the Config scenario, we do not think the `@retry` decorator suits this case (but it's precious :sparkles:). And we think `solve_problem` function should not modify the config because it is initialized and mostly used outside the function. As Benji would say, do we accept octopus? :octopus: